src/orchestratord/src/controller/materialize/resources.rs

```diff
@@ -899,13 +871,6 @@ fn create_environmentd_statefulset_object(
         config.secrets_controller
     ));
 
-    if config.local_development {
-        args.extend([
-            "--system-parameter-default=cluster_enable_topology_spread=false".into(),
```

Comment:
one note: this breaks the defaults for local kube installations, all of kind / minikube / k3d by default do not set the `topology.kubernetes.io/zone` label, so clusterd gets jammed scheduling:

```
Topology Spread Constraints:  topology.kubernetes.io/zone:DoNotSchedule when max skew 1 is exceeded for selector cluster.environmentd.materialize.cloud/cluster-id in (u1),cluster.environmentd.materialize.cloud/scale in (1)
Events:
  Type     Reason            Age    From               Message
  ----     ------            ----   ----               -------
  Warning  FailedScheduling  3m44s  default-scheduler  0/1 nodes are available: 1 node(s) didn't match pod topology spread constraints (missing required label). preemption: 0/1 nodes are available: 1 Preemption is not helpful for scheduling.
```

This might be fine and we might just need to call it out in the docs, but I did want to note it

cc @kay-kim if we merge as-is we'll need to add a new knob for running locally that sets `--system-parameter-default=cluster_enable_topology_spread=false`

Reply:
oh, yet another thing i missed by using the cloud kind cluster (which does explicitly set the zone) - i really should stop using that for testing this stuff

Reply:
i just readded the flag if the cloud provider they passed corresponds to local development, which i think should be fine for now?

Reply:
oh sweet, that's perfect 👍